ConcernsCaseWork/Service.Redis.Tests/Teams/TeamsCachedServiceTests.cs

```diff
@@ -45,5 +42,92 @@ public void Constructors_GuardAgainstNullArgs()
 			var assertion = fixture.Create<GuardClauseAssertion>();
 			assertion.Verify(typeof(TeamsCachedService).GetConstructors());
 		}
+
+		[Test]
+		public async Task GetTeam_When_NotCached_Returns_Data_From_TeamsService()
+		{
+			// arrange
+			var expectedTeamOwner = "john.smith";
+			var mockCacheProvider = new Mock<ICacheProvider>();
+
+			var mockTeamsService = new Mock<ITeamsService>();
+			mockTeamsService.Setup(x => x.GetTeam(expectedTeamOwner)).ReturnsAsync(new ConcernsCaseworkTeamDto(expectedTeamOwner, Array.Empty<string>()));
+
+			var sut = new TeamsCachedService(Mock.Of<ILogger<TeamsCachedService>>(), mockTeamsService.Object, mockCacheProvider.Object);
+
+			// act
+			var team = await sut.GetTeam(expectedTeamOwner);
+
+			// assert
+			Assert.IsNotNull(team);
+		}
+
+		[Test]
+		public async Task GetTeam_When_Cached_Returns_CachedData()
+		{
+			// arrange
+			var expectedTeamOwner = "john.smith";
+			var expectedCacheKey = "Concerns.Teams.john.smith";
+			var mockCacheProvider = new Mock<ICacheProvider>();
+			mockCacheProvider.Setup(x => x.GetFromCache<ConcernsCaseworkTeamDto>(expectedCacheKey))
+				.ReturnsAsync(new ConcernsCaseworkTeamDto(expectedCacheKey, Array.Empty<string>()));
+
+			var mockTeamsService = new Mock<ITeamsService>();
+
+			var sut = new TeamsCachedService(Mock.Of<ILogger<TeamsCachedService>>(), mockTeamsService.Object, mockCacheProvider.Object);
+
+			// act
+			var team = await sut.GetTeam(expectedTeamOwner);
+
+			// assert
+			Assert.IsNotNull(team);
+		}
+
+		[Test]
+		public async Task GetTeamOwners_ByPasses_Cache()
+		{
+			// arrange
+			var expectedTeamOwner = "john.smith";
+			var mockCacheProvider = new Mock<ICacheProvider>(MockBehavior.Strict);
+
+			var mockTeamsService = new Mock<ITeamsService>();
+			mockTeamsService.Setup(x => x.GetTeamOwners()).ReturnsAsync(new[] { expectedTeamOwner });
+
+			var sut = new TeamsCachedService(Mock.Of<ILogger<TeamsCachedService>>(), mockTeamsService.Object, mockCacheProvider.Object);
+
+			// act
+			var teamOwners = await sut.GetTeamOwners();
+
+			// assert
+			Assert.That(teamOwners.Length == 1);
```

Comment:
I'm not sure if Assert.Equal gives a better error message - I think it's recommended over using assert.that anyway. (or use It.Is format)